extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerRecorder.java

```diff
@@ -27,24 +17,13 @@ public class TracerRecorder {
     public static final Set<String> dropStaticResourceTargets = new HashSet<>();
 
     /* STATIC INIT */
-    public RuntimeValue<Optional<IdGenerator>> createIdGenerator() {
-        BeanManager beanManager = Arc.container().beanManager();
-
-        Instance<IdGenerator> idGenerator = beanManager.createInstance()
```

Comment:
This might break idGeneration. We don't have a proper test. Will come up with something, to make sure.

Reply:
Wouldn't we already have this problem in the current version?